app/controllers/candidate_interface/equality_and_diversity_controller.rb

```diff
@@ -146,6 +138,28 @@ def ethnic_background_param
       params.require(:candidate_interface_equality_and_diversity_ethnic_background_form).permit(:ethnic_background, :other_background)
     end
 
+    def free_school_meals_param
+      params.require(:candidate_interface_equality_and_diversity_free_school_meals_form).permit(:free_school_meals)
+    end
+
+    def free_school_meals_or_review(application)
+      if application.ask_about_free_school_meals?
+        candidate_interface_edit_equality_and_diversity_free_school_meals_path
+      else
+        candidate_interface_review_equality_and_diversity_path
+      end
+    end
+
+    def check_that_candidate_should_be_asked_about_free_school_meals
+      render_404 unless current_application.ask_about_free_school_meals?
```

Comment:
Could redirect to the review page here, instead of 404ing, just in case? (Unlikely to happen though!)